app/models/manageiq/providers/ibm_power_hmc/infra_manager/event_catcher/stream.rb

```diff
@@ -17,19 +14,28 @@ def stop
     @stop_polling = true
   end
 
-  def poll
+  def poll_events(connection, &block)
+    loop do
+      start_time = Time.now.utc.to_i
+      events = connection.next_events(false)
+      elapsed = Time.now.utc.to_i - start_time
+
+      events.each(&block)
+      break if @stop_polling
+
+      # The HMC already waits a fair amount of time before returning
+      # if there were no events pending, so substract the query time
+      # from the poll sleep interval.
+      sleep(@poll_sleep - elapsed) if @poll_sleep > elapsed
```

Comment:
This is fine but just curious how long does HMC tend to wait?

Reply:
Reason I'm asking is if it blocks for long enough we might not need the sleep at all

Reply:
Based on observations on several HMCs, the HMC waits for 10 seconds before it returns in case of no events (covering for Damien today)

Reply:
Yes, it is 10 seconds:

```
2.7.2 :008 > Benchmark.measure { hc.next_events(false) }
 => #<Benchmark::Tms:0x00000100092c86e8 @label="", @real=10.05330021888949, @cstime=0.0, @cutime=0.0, @stime=0.0, @utime=0.012647000000000297, @total=0.012647000000000297>
2.7.2 :009 > Benchmark.measure { hc.next_events(false) }
 => #<Benchmark::Tms:0x000001000924bb70 @label="", @real=10.039398250170052, @cstime=0.0, @cutime=0.0, @stime=0.0, @utime=0.012808999999999848, @total=0.012808999999999848>
2.7.2 :010 > Benchmark.measure { hc.next_events(false) }
 => #<Benchmark::Tms:0x00000100091cec10 @label="", @real=10.04056040220894, @cstime=0.0, @cutime=0.0, @stime=0.01261799999999999, @utime=0.0, @total=0.01261799999999999>
```

Reply:
Interestingly, the HMC supports a hidden "timeout" option that maybe we could set to @poll_sleep:

```
2.7.2 :015 > Benchmark.measure { hc.request(:get, "/rest/api/uom/Event") }.real
 => 10.065891639096662
2.7.2 :016 > Benchmark.measure { hc.request(:get, "/rest/api/uom/Event?timeout=1") }.real
 => 1.0571004541125149
2.7.2 :014 > Benchmark.measure { hc.request(:get, "/rest/api/uom/Event?timeout=2") }.real
 => 2.05112814810127
2.7.2 :013 > Benchmark.measure { hc.request(:get, "/rest/api/uom/Event?timeout=5") }.real
 => 5.064436040818691
```

Reply:
:+1: yeah up to you guys but this would be greatly cleaned up/simplified if we dropped the extra sleep

Reply:
I'll add a timeout option to the SDK.